app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -1,6 +1,7 @@
 module Api
   class AutomateWorkspacesController < BaseController
     def edit_resource(type, id, data = {})
+      raise BadRequestError, "must contain at least one attribute to edit" if data.blank?
```

Comment:
@mkanoor is this the only case from the model that was raising the ArgumentError ?